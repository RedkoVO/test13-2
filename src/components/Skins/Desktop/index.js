import React from 'react'
import PropTypes from 'prop-types'
import withStyles from '@material-ui/core/styles/withStyles'

import MenuNavigation from '../../MenuNavigation/Desktop/AsyncMenuNavigation'
import Header from '../../../containers/Header/AsyncHeader'
import TradeTop from '../components/TradeTop'
import TradeSkins from '../components/TradeSkins'
import TradePayment from '../components/TradePayment'
import Footer from '../../Footer/Desktop/'

import styles from './styles'

const Skins = ({ classes, handleLogout, skins, selectedItems }) => (
  <div className="mainWrapperTrade">
    <div className={classes.containerTradeWrap}>
      <div className={classes.containerTrade}>
        <header className={classes.headerSkins}>
          <MenuNavigation type="main" page="dark" handleLogout={handleLogout} />
          <Header type="catalog" />
        </header>

        <main className={classes.tradeMain}>
          <TradeTop />
          <TradeSkins skins={skins} />
          <TradePayment selectedItems={selectedItems} />
        </main>
      </div>
    </div>

    <Footer />
  </div>
)

Skins.propTypes = {
  classes: PropTypes.object,
  handleLogout: PropTypes.func,
  skins: PropTypes.array,
  selectedItems: PropTypes.array
}

export default withStyles(styles)(Skins)
