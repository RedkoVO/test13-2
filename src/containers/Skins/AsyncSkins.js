import Async from '../../components/App/components/Async'

export const AsyncSkins = Async({
  loader: () => import(/* webpackChunkName: "skins_page" */ './index')
})

export default AsyncSkins
