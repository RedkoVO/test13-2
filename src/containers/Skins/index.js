import compose from 'recompose/compose'
import { withProps, pure } from 'recompose'

import AsyncSkinsDesktop from '../../components/Skins/Desktop/AsyncSkinsDesktop'

const tmpArrSkins = [
  {
    id: 1,
    title: 'Gun: Good Weapon2 1.2',
    cost: '$11.10 FN',
    discountsCost: '$12.30'
  },
  {
    id: 2,
    title: 'Gun: Good Weapon2 1.2',
    cost: '$11.10 FN',
    discountsCost: '$12.30'
  },
  {
    id: 3,
    title: 'Gun: Good Weapon2 1.2',
    cost: '$11.10 FN',
    discountsCost: '$12.30'
  },
  {
    id: 4,
    title: 'Gun: Good Weapon2 1.2',
    cost: '$11.10 FN',
    discountsCost: '$12.30'
  },
  {
    id: 5,
    title: 'Gun: Good Weapon2 1.2',
    cost: '$11.10 FN',
    discountsCost: '$12.30'
  },
  {
    id: 6,
    title: 'Gun: Good Weapon2 1.2',
    cost: '$11.10 FN',
    discountsCost: '$12.30'
  },
  {
    id: 7,
    title: 'Gun: Good Weapon2 1.2',
    cost: '$11.10 FN',
    discountsCost: '$12.30'
  },
  {
    id: 8,
    title: 'Gun: Good Weapon2 1.2',
    cost: '$11.10 FN',
    discountsCost: '$12.30'
  },
  {
    id: 9,
    title: 'Gun: Good Weapon2 1.2',
    cost: '$11.10 FN',
    discountsCost: '$12.30'
  },
  {
    id: 10,
    title: 'Gun: Good Weapon2 1.2',
    cost: '$11.10 FN',
    discountsCost: '$12.30'
  },
  {
    id: 11,
    title: 'Gun: Good Weapon2 1.2',
    cost: '$11.10 FN',
    discountsCost: '$12.30'
  },
  {
    id: 12,
    title: 'Gun: Good Weapon2 1.2',
    cost: '$11.10 FN',
    discountsCost: '$12.30'
  }
]
const selectedItems = [
  { id: 1, title: 'Good EWeapon 1.2', cost: '$12.20' },
  { id: 2, title: 'Good EWeapon 1.2', cost: '$12.20' },
  { id: 3, title: 'Good EWeapon 1.2', cost: '$12.20' },
  { id: 4, title: 'Good EWeapon 1.2', cost: '$12.20' },
  { id: 5, title: 'Good EWeapon 1.2', cost: '$12.20' },
  { id: 6, title: 'Good EWeapon 1.2', cost: '$12.20' }
]

export default compose(
  withProps(() => ({ skins: tmpArrSkins, selectedItems: selectedItems })),
  pure
)(AsyncSkinsDesktop)
