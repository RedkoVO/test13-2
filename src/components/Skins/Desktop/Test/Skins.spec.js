import * as React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'

import Skins from '../'

const tmpArrSkins = [
  {
    id: 1,
    title: 'Gun: Good Weapon2 1.2',
    cost: '$11.10 FN',
    discountsCost: '$12.30'
  },
  {
    id: 2,
    title: 'Gun: Good Weapon2 1.2',
    cost: '$11.10 FN',
    discountsCost: '$12.30'
  },
  {
    id: 3,
    title: 'Gun: Good Weapon2 1.2',
    cost: '$11.10 FN',
    discountsCost: '$12.30'
  }
]
const selectedItems = [
  { id: 1, title: 'Good EWeapon 1.2', cost: '$12.20' },
  { id: 2, title: 'Good EWeapon 1.2', cost: '$12.20' },
  { id: 3, title: 'Good EWeapon 1.2', cost: '$12.20' },
  { id: 4, title: 'Good EWeapon 1.2', cost: '$12.20' },
  { id: 5, title: 'Good EWeapon 1.2', cost: '$12.20' },
  { id: 6, title: 'Good EWeapon 1.2', cost: '$12.20' }
]

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter keyLength={0}>
        <Skins
          handleLogout={() => {}}
          isDesktop={false}
          skins={tmpArrSkins}
          selectedItems={selectedItems}
        />
      </MemoryRouter>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
