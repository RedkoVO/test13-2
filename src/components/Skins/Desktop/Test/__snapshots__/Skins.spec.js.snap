// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mainWrapperTrade"
>
  <div
    className="Skins-containerTradeWrap-1"
  >
    <div
      className="Skins-containerTrade-2"
    >
      <header
        className="Skins-headerSkins-3"
      />
      <main
        className="Skins-tradeMain-4"
      >
        <div
          className="TradeTop-root-5"
        >
          <div
            className="TradeTop-tradeGame-6"
          >
            <div
              className="TradeTop-tradeGameSelect-7"
            >
              Game:
            </div>
            <div
              className="TradeTop-tradeGameItem-8"
            >
              CS:GO
            </div>
          </div>
          <div
            className="TradeTop-tradeStream-9"
          >
            <div
              className="TradeTop-tradeStreamTitle-10"
            >
              Put your steam trade url: 
            </div>
            <div
              className="TradeTop-tradeStreamUrlWrap-11"
            >
              <input
                className="TradeTop-tradeStreamUrl-12"
                placeholder="https://steamcommunity.com/groups/dota2lounge/discussions/0/558"
                type="text"
              />
            </div>
          </div>
        </div>
        <div
          className="TradeSkins-root-13"
        >
          <div
            className="TradeSkins-tradeSkinsTitle-14"
          >
            Select skins you want to sale:
          </div>
          <div
            className="TradeSkins-tradeSkinsItemWrap-15"
          >
            <div
              className="TradeSkins-tradeSkinsItem-16 TradeSkins-skinSelected-17"
            >
              <div
                className="TradeSkins-skinsItemImg-18"
              >
                <img
                  alt=""
                  src="gun.png"
                />
              </div>
              <div
                className="TradeSkins-skinsItemName-19"
              >
                Gun: Good Weapon2 1.2
              </div>
              <div
                className="TradeSkins-skinsItemPrice-20"
              >
                $11.10 FN
              </div>
              <div
                className="TradeSkins-skinsItemSale-21"
              >
                $12.30
              </div>
            </div>
            <div
              className="TradeSkins-tradeSkinsItem-16 TradeSkins-skinSelected-17"
            >
              <div
                className="TradeSkins-skinsItemImg-18"
              >
                <img
                  alt=""
                  src="gun.png"
                />
              </div>
              <div
                className="TradeSkins-skinsItemName-19"
              >
                Gun: Good Weapon2 1.2
              </div>
              <div
                className="TradeSkins-skinsItemPrice-20"
              >
                $11.10 FN
              </div>
              <div
                className="TradeSkins-skinsItemSale-21"
              >
                $12.30
              </div>
            </div>
            <div
              className="TradeSkins-tradeSkinsItem-16 TradeSkins-skinSelected-17"
            >
              <div
                className="TradeSkins-skinsItemImg-18"
              >
                <img
                  alt=""
                  src="gun.png"
                />
              </div>
              <div
                className="TradeSkins-skinsItemName-19"
              >
                Gun: Good Weapon2 1.2
              </div>
              <div
                className="TradeSkins-skinsItemPrice-20"
              >
                $11.10 FN
              </div>
              <div
                className="TradeSkins-skinsItemSale-21"
              >
                $12.30
              </div>
            </div>
          </div>
          <ul
            className="TradeSkins-pageNavi-22"
          >
            <li>
              <div
                className="TradeSkins-previos-23"
              >
                Prev
              </div>
            </li>
            <li>
              <div>
                1
              </div>
            </li>
            <li>
              <div>
                2
              </div>
            </li>
            <li>
              <div
                className="TradeSkins-next-24"
              >
                Next
              </div>
            </li>
          </ul>
        </div>
        <div
          className="TradePayment-root-25"
        >
          <div
            className="TradePayment-paymentTitle-26"
          >
            Choose the way to cash out:
          </div>
          <div
            className="TradePayment-paymentWay-27"
          >
            <div
              className="TradePayment-paymentWayItem-28 TradePayment-paymentSelectItem-29"
            >
              <div
                className="TradePayment-paymentWayItemImg-30"
              >
                <img
                  alt=""
                  src="visa.png"
                />
              </div>
              <div
                className="TradePayment-paymentWayItemName-31"
              >
                Credit Card
              </div>
              <div
                className="TradePayment-paymentWayItemUnderline-32 TradePayment-selected-33"
              />
            </div>
            <div
              className="TradePayment-paymentWayItem-28"
            >
              <div
                className="TradePayment-paymentWayItemImg-30"
              >
                <img
                  alt=""
                  src="bitcnpay.png"
                />
              </div>
              <div
                className="TradePayment-paymentWayItemName-31"
              >
                Bitcoin
              </div>
              <div
                className="TradePayment-paymentWayItemUnderline-32"
              />
            </div>
            <div
              className="TradePayment-paymentWayItem-28"
            >
              <div
                className="TradePayment-paymentWayItemImg-30"
              >
                <img
                  alt=""
                  src="agspay.png"
                />
              </div>
              <div
                className="TradePayment-paymentWayItemName-31"
              >
                CryptaurCoin
              </div>
              <div
                className="TradePayment-paymentWayItemUnderline-32"
              />
            </div>
          </div>
          <div
            className="PaymentCard-root-44"
          >
            <div
              className="PaymentCard-paymentCardTitle-45"
            >
              Cashout on my Credit card:
            </div>
            <div
              className="PaymentCard-paymentInputWrap-46"
            >
              <input
                className="PaymentCard-paymentCardNum-48"
                placeholder="Card Number"
                type="text"
              />
            </div>
            <div
              className="PaymentCard-paymentInputWrap2-47"
            >
              <input
                className="PaymentCard-paymentCardHolder-49"
                placeholder="Card Holder"
                type="text"
              />
            </div>
            <div
              className="PaymentCard-paymentCardMonth-50"
            >
              MM
            </div>
            <div
              className="PaymentCard-paymentCardYear-51"
            >
              YYYY
            </div>
            <div
              className="PaymentCard-paymentInputWrap3-52"
            >
              <input
                className="PaymentCard-paymentCardCvc-53"
                placeholder="CVC"
                type="text"
              />
            </div>
          </div>
          <div
            className="PaymentSelected-root-54"
          >
            <div
              className="PaymentSelected-paymentSelectedTitle-55"
            >
              You selected 6 items:
            </div>
            <div
              className="PaymentSelected-paymentSelectedItem-56"
            >
              <div>
                0
                ) 
                Good EWeapon 1.2
                 ....
              </div>
              <div
                className="PaymentSelected-selectedPrice-57"
              >
                $12.20
              </div>
              <div
                className="PaymentSelected-selectedClose-58"
              />
            </div>
            <div
              className="PaymentSelected-paymentSelectedItem-56"
            >
              <div>
                1
                ) 
                Good EWeapon 1.2
                 ....
              </div>
              <div
                className="PaymentSelected-selectedPrice-57"
              >
                $12.20
              </div>
              <div
                className="PaymentSelected-selectedClose-58"
              />
            </div>
            <div
              className="PaymentSelected-paymentSelectedItem-56"
            >
              <div>
                2
                ) 
                Good EWeapon 1.2
                 ....
              </div>
              <div
                className="PaymentSelected-selectedPrice-57"
              >
                $12.20
              </div>
              <div
                className="PaymentSelected-selectedClose-58"
              />
            </div>
            <div
              className="PaymentSelected-paymentSelectedItem-56"
            >
              <div>
                3
                ) 
                Good EWeapon 1.2
                 ....
              </div>
              <div
                className="PaymentSelected-selectedPrice-57"
              >
                $12.20
              </div>
              <div
                className="PaymentSelected-selectedClose-58"
              />
            </div>
            <div
              className="PaymentSelected-paymentSelectedItem-56"
            >
              <div>
                4
                ) 
                Good EWeapon 1.2
                 ....
              </div>
              <div
                className="PaymentSelected-selectedPrice-57"
              >
                $12.20
              </div>
              <div
                className="PaymentSelected-selectedClose-58"
              />
            </div>
            <div
              className="PaymentSelected-paymentSelectedItem-56"
            >
              <div>
                5
                ) 
                Good EWeapon 1.2
                 ....
              </div>
              <div
                className="PaymentSelected-selectedPrice-57"
              >
                $12.20
              </div>
              <div
                className="PaymentSelected-selectedClose-58"
              />
            </div>
          </div>
          <div
            className="TradePayment-paymentAmmount-34"
          >
            <div
              className="TradePayment-paymentAmmountText-35"
            >
              Total ammount: 
            </div>
            <div
              className="TradePayment-paymentAmmountFill-36"
            />
            <div
              className="TradePayment-paymentAmmountPrice-37"
            >
              $67.45
            </div>
          </div>
          <div
            className="TradePayment-paymentBonus-38"
          >
            <input
              className="TradePayment-paymentBonusCheck-39"
              id="paymentBonusCheck"
              type="checkbox"
            />
            <label
              className="TradePayment-paymentBonusCheckLabel-40"
              htmlFor="paymentBonusCheck"
            >
              Yes, I have a bonus code!
            </label>
          </div>
          <div
            className="TradePayment-paymentInputWrap4-41"
          >
            <input
              className="TradePayment-paymentBonusInput-42"
              placeholder="Enter Your Bonus Code"
              type="text"
            />
          </div>
          <div
            className="TradePayment-paymentBtn-43"
          >
            CASHOUT
          </div>
        </div>
      </main>
    </div>
  </div>
  <footer
    className="Footer-root-59"
  >
    <ul
      className="Footer-footerNav-60"
    >
      <li>
        <a
          className="Footer-footerNavItem-61"
          href="/"
          onClick={[Function]}
        >
          Dashboard
        </a>
      </li>
      <li>
        <a
          className="Footer-footerNavItem-61"
          href="/"
          onClick={[Function]}
        >
          Wallet
        </a>
      </li>
      <li>
        <a
          className="Footer-footerNavItem-61"
          href="/"
          onClick={[Function]}
        >
          Shop
        </a>
      </li>
      <li>
        <a
          className="Footer-footerNavItem-61"
          href="/games"
          onClick={[Function]}
        >
          Games
        </a>
      </li>
      <li>
        <a
          className="Footer-footerNavItem-61"
          href="/"
          onClick={[Function]}
        >
          Friends
        </a>
      </li>
      <li>
        <a
          className="Footer-footerNavItem-61"
          href="/"
          onClick={[Function]}
        >
          Settings
        </a>
      </li>
    </ul>
    <span>
      Vault. All Rights Reserved 2018
    </span>
  </footer>
</div>
`;
